tests/jenkins/TestDockerCopy.groovy

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+import jenkins.tests.BuildPipelineTest
+import org.junit.Before
+import org.junit.Test
+import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
+import static org.hamcrest.CoreMatchers.equalTo
+import static org.hamcrest.CoreMatchers.hasItem
+import static org.hamcrest.CoreMatchers.hasItems
+import static org.hamcrest.CoreMatchers.notNullValue
+import static org.hamcrest.MatcherAssert.assertThat
+
+class TestDockerCopy extends BuildPipelineTest {
+
+    String sourceImageRegistry = 'opensearchstaging'
+    String sourceImage = 'opensearch:1.3.2'
+    String destinationImageRegistry = 'opensearchproject'
+    String destinationImage = 'opensearch:1.3.2'
+
+    @Override
+    @Before
+    void setUp() {
+
+        super.setUp()
+
+        // Variables
+        binding.setVariable('SOURCE_IMAGE_REGISTRY', sourceImageRegistry)
+        binding.setVariable('SOURCE_IMAGE', sourceImage)
+        binding.setVariable('DESTINATION_IMAGE_REGISTRY', destinationImageRegistry)
+        binding.setVariable('DESTINATION_IMAGE', destinationImage)
+
+    }
+
+    @Test
+    void DockerCopyRegression() {
+        super.testPipeline('jenkins/docker/docker-copy.jenkinsfile',
+                'tests/jenkins/jenkinsjob-regression-files/docker/docker-copy.jenkinsfile')
+    }
+
+    @Test
+    public void DockerCopyExecuteWithoutErrors() {
+        runScript("jenkins/docker/docker-copy.jenkinsfile")
+
+        assertJobStatusSuccess()
+
+        // Ensure the gcrane is executed in an external shell script exactely once
+        def copyContainerCommand = getCommands('docker').findAll {
+            shCommand -> shCommand.contains('gcrane cp opensearchstaging/opensearch:1.3.2 opensearchproject/opensearch:1.3.2; docker logout')
+        }
+        assertThat(copyContainerCommand.size(), equalTo(1))
+
+        // Validate the copyContainer docker-copy.sh is called
+        assertCallStack().contains("docker-copy.copyContainer")
```

Comment:
I don't think we have anything called as `docker-copy.sh` to validate? Did you mean to check if the callStack contains `copyContainer.sh`?

Reply:
Hi, @gaiksaya , thanks for the question, I think the first part, docker-copy, is the docker-copy.jenkinsfile, and the 2nd part, copyContainer.groovy, is called by docker-copy.jenkinsfile.
I thought it would be great to validate both in one assertion. :) 
So, we make sure the jenkinsfile does call the copyContainer lib.
If otherwise, I can remove it as well.. please feel free to let me know. 
thanks,

Reply:
Can you fix the comment above saying `// Validate the copyContainer.sh is called by docker-copy` thats why I got confused

Reply:
oops... my bad ... I did not mean to say .sh in the comment.. LOL
I am correcting it now.. thanks for pointing it out!